src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_UsingStatement.cs

```diff
@@ -35,40 +35,75 @@ public override BoundNode VisitUsingStatement(BoundUsingStatement node)
-                Conversion idisposableConversion = node.IDisposableConversion;
-                ImmutableArray<BoundLocalDeclaration> declarations = node.DeclarationsOpt.LocalDeclarations;
+        private BoundStatement MakeDeclarationUsingStatement(SyntaxNode syntax,
+                                                       BoundBlock body, 
+                                                       ImmutableArray<LocalSymbol> locals,
+                                                       ImmutableArray<BoundLocalDeclaration> declarations,
+                                                       Conversion iDisposableConversion,
+                                                       MethodSymbol disposeMethodOpt,
+                                                       AwaitableInfo awaitOpt,
+                                                       SyntaxToken awaitKeyword)
+        {
+            Debug.Assert(declarations != null);
 
-                BoundBlock result = tryBlock;
+            BoundBlock result = body;
+            for (int i = declarations.Length - 1; i >= 0; i--) //NB: inner-to-outer = right-to-left
+            {
+                result = RewriteDeclarationUsingStatement(syntax, declarations[i], result, iDisposableConversion, awaitKeyword, awaitOpt, disposeMethodOpt);
+            }
 
-                int numDeclarations = declarations.Length;
-                for (int i = numDeclarations - 1; i >= 0; i--) //NB: inner-to-outer = right-to-left
-                {
-                    result = RewriteDeclarationUsingStatement(usingSyntax, declarations[i], result, idisposableConversion, awaitKeyword, node.AwaitOpt, node.DisposeMethodOpt);
-                }
+            // Declare all locals in a single, top-level block so that the scope is correct in the debugger
+            // (Dev10 has them all come into scope at once, not per-declaration.)
+            return new BoundBlock(
+                syntax,
+                locals,
+                ImmutableArray.Create<BoundStatement>(result));
+        }
 
-                // Declare all locals in a single, top-level block so that the scope is correct in the debugger
-                // (Dev10 has them all come into scope at once, not per-declaration.)
-                return new BoundBlock(
-                    usingSyntax,
-                    node.Locals,
-                    ImmutableArray.Create<BoundStatement>(result));
-            }
+        /// <summary>
+        /// Lower "using var x = (expression)" to a try-finally block.
+        /// </summary>
+        private BoundStatement MakeLocalUsingDeclarationStatement(BoundUsingLocalDeclarations usingDeclarations, ImmutableArray<BoundStatement> statements)
+        {
+            SyntaxNode syntax = usingDeclarations.Syntax;
+            BoundBlock body = new BoundBlock(syntax, ImmutableArray<LocalSymbol>.Empty, statements);
```

Comment:
>BoundBlock [](start = 34, length = 10)

Is there a `BoundBlock` that contains the locals from the declaration?

Reply:
Yes, the original parent block it was declared in. All the synthesized blocks generated as part of lowering the using declarations have no locals, and they all belong to the original block.